### PR TITLE
Added authorize + fetch-transaction classes

### PR DESCRIPTION
**Pull request by fred-jan**

### CODE REVIEW

File: src/Gateway.php [hunk 1]
```diff
@@ -78,6 +80,27 @@ public function getSecret()
...
+     * @inheritDoc
+     */
+    public function initialize(array $parameters = array())
+    {
+        parent::initialize($parameters);
+
+        if (self::API_VERSION_EUROPE === $this->getApiRegion()) {
+            $baseUrl = $this->getTestMode() ? ConnectorInterface::EU_TEST_BASE_URL : ConnectorInterface::EU_BASE_URL;
+        } else {
+            $baseUrl = $this->getTestMode() ? ConnectorInterface::NA_BASE_URL : ConnectorInterface::NA_TEST_BASE_URL;
```
Comment (Feijs) [inline src/Gateway.php]: Test and base constants should be swapped (test is also incorrect, sorry)

File: src/Item.php [hunk 2]
```diff
@@ -13,10 +13,26 @@ public function getTaxRate()
     }
 
     /**
+     * @inheritDoc
+     */
+    public function getTotalTaxAmount()
```
Comment (Feijs) [inline src/Item.php:18]: This getters/setter combo is never used. Item transformation uses a calculated value `$item->getQuantity() * $item->getPrice() * 100`

File: src/Message/AuthorizeRequest.php [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+final class AuthorizeRequest extends AbstractRequest
+{
+    use ItemDataTrait;
+
+    /**
+     * @inheritDoc
+     */
+    public function getData()
+    {
+        $this->validate('terms_url');
```
Comment (Feijs) [inline src/Message/AuthorizeRequest.php]: Optional: terms_url is not the only required parameter, the presence of more parameters could be asserted here

**Review by Feijs — CHANGES_REQUESTED** [review]

File: src/Item.php (hunk 2 above)
Comment (fred-jan) [reply src/Item.php:18]: This parameter is actually used in `ItemDataTrait::getItemData()`: `'total_tax_amount' => $item->getTotalTaxAmount() * 100,`

**Review by Feijs — APPROVED** [review]

**Pull request merged**